packages/node-core/src/indexer/blockDispatcher/base-block-dispatcher.ts

```diff
@@ -187,6 +187,13 @@ export abstract class BaseBlockDispatcher<Q extends IQueue, DS, B> implements IB
           void this.poiSyncService.syncPoi();
         }
         this.eventEmitter.emit(IndexerEvent.RewindSuccess, {success: true, height: reindexBlockHeader.blockHeight});
+
+        // End transaction
+        await this.storeModelProvider.applyPendingChanges(
+          height,
+          !this.projectService.hasDataSourcesAfterHeight(height),
+          this.storeService.transaction
+        );
```

Comment:
Should this happen before the rewind? It seems wrong to apply cache changes, after rewind.